### DISCUSSION TITLE
Ignored Build Step: Git diff command returns code 1 on my machine but vercel build is still skipped

### DISCUSSION THREAD

**Discussion by baptisteArno** [vercel/vercel#5676] [Help]
Hi 👋

I'd like to trigger the deploy process only if a subdirectory contains changes. So I changed the diff command in Vercel settings with this:

```sh
git diff --quiet HEAD^ HEAD -- apps/landing-page  
```

But it still skip deploy even if `landing-page` contains changes. Running the command on my machine returns 1 and I suppose it returns 0 on the vercel machine? Am I missing something?

**Comment by rainvou** [comment]
@baptisteArno 
Hi  
Did you solve it?
If you solved it.. I want know....  plz~

**Comment by baptisteArno** [comment]
Still not solved...

**Comment by colophonemes** [comment]
Adding my voice to this. I can't get any form of `git diff HEAD^ HEAD --quiet -- ./packages/myapp ./packages/common ./packages/graphql` working (have tried numerous variations). `echo $?` returns `1` on my local machine after running the line.

**Reply by colophonemes** [reply]
@styfle @leerob @mcsdevv @leo any ideas? Sorry to be pushy, but this is causing us to run twice as many builds (and for CI to take ~twice as long to complete) every time we push to the repo.

**Reply by styfle** [reply]
Try `../packages/myapp` since the command is run in the [Root Directory](https://vercel.com/docs/build-step#root-directory) you defined for the current project.

**Reply by colophonemes** [reply]
Amazing, that works, thank you!

This was a bit non-obvious to me (perhaps it should have been), so I wonder if it's worth adding a line to [the docs page](https://vercel.com/support/articles/how-do-i-use-the-ignored-build-step-field-on-vercel) or to the helper text in the dashboard. The docs sort of allude to the issue in one of the examples, but might be good to spell out explicitly.

**Reply by styfle** [reply]
Good idea! We'll update that page, thanks! 👍

**Reply by baptisteArno** [reply]
> Try `../packages/myapp` since the command is run in the [Root Directory](https://vercel.com/docs/build-step#root-directory) you defined for the current project.

Ok, so it works fine when I set `git diff --quiet HEAD^ HEAD -- .`. Maybe it should be the default behaviour?
